File: csharp/src/Structural/Proxy/PotOfTeaProxy.cs
namespace Structural.Proxy {


//PotOfTeaProxy.java - the Proxy

public class PotOfTeaProxy : PotOfTeaInterface {  
   PotOfTea potOfTea;
    
   public PotOfTeaProxy() {}
   
   public string pourTea() {
       potOfTea = new PotOfTea();
       return potOfTea.pourTea();
   }
}

}

File: go/structural/proxy/PotOfTeaProxy.go
package proxy


//PotOfTeaProxy - the Proxy

type PotOfTeaProxy struct {} //implements PotOfTeaInterface {  
    
func (this *PotOfTeaProxy) PourTea() string {
	 potOfTea := new(PotOfTea)
	 return potOfTea.PourTea()
}
